Dockerfile

```diff
@@ -17,18 +17,21 @@ COPY crates crates
 COPY cairo cairo
 COPY cairo_0 cairo_0
 
-# Installing scarb, new since devnet integration
-# Installation steps are taken from the scarb build script
-# https://github.com/software-mansion/scarb/blob/main/install.sh
-ENV SCARB_VERSION="v2.8.2"
-ENV SCARB_REPO="https://github.com/software-mansion/scarb/releases/download"
-ENV PLATFORM="x86_64-unknown-linux-gnu"
-ENV SCARB_TARGET="/usr/src/scarb.tar.gz"
-
-RUN curl -fLS -o $SCARB_TARGET                                          \
-    $SCARB_REPO/$SCARB_VERSION/scarb-$SCARB_VERSION-$PLATFORM.tar.gz && \
-    tar -xz -C /usr/src/ --strip-components=1 -f $SCARB_TARGET &&       \
-    mv /usr/src/bin/scarb /bin
+# Installing Scarb
+# Dynamically detect the architecture
```

Comment:
Why are we checking for architecture inside of a Dockerfile? This will always resolve to whatever image we are building this under and should be set to a fixed architecture instead.

Reply:
for some reason this error was thrown on mac builds:

![image](https://github.com/user-attachments/assets/f27c3186-dd4e-40df-8fa8-113d65b61bc1)